docs/source/daml/intro/4_Transformations.rst

```diff
@@ -0,0 +1,154 @@
+
+There is nothing here indicating that the current ``Contact`` should be archived. That's because choices are *consuming* by default. That means when the above choice is exercised on a contract, that contract is archived.
+
+If you paid a lot of attention in :doc:`3_Data`, you may have noticed that the ``create`` statement returns an ``Update (ContractId Contact)``, not a ``ContractId Contact``. As a ``do`` block always returns the value of the last statement within it, the whole ``do`` block returns an ``Update``, but the return type on the choice is just a ``ContractId Contact``. This is a conveneience. Choices *always* return an ``Update`` so for readability it's omitted on the type declaration of a choice.
+
+Now to exercise the new choice in a scenario:
+
+.. literalinclude:: daml/4_Transformations/Contact.daml
+  :language: daml
+  :start-after: -- CHOICE_TEST_BEGIN
+  :end-before: -- CHOICE_TEST_END
+
+Choices are exercised using the ``exercise`` function, which takes a ``ContractId a``, and a value of type ``c``, where ``c`` is a choice on template ``a``. Since ``c`` is just a record, you can also just fill in the choice parameters using the ``with`` syntax you are already familiar with. ``exercise`` returns an ``Update r`` where ``r`` is the return type specified on the choice, allowing the new ``ContractId Contact`` to be stored in the variable ``new_acontactCid``.
+
+Choices as Delegation
+---------------------
+
+Up to this point all the contracts only involved one party. ``party`` may have been stored as ``Party`` field in the above, which suggests they are actors on the ledger, but they couldn't see the contracts, nor change them in any way. It would be reasonable for the party for which a ``Contact`` is stored to be able to update their own address and telephone number. The below demonstrates this using an ``UpdateAddress`` choice and corresponding extension of the scenario:
+
+.. literalinclude:: daml/4_Transformations/Contact.daml
+  :language: daml
+  :start-after: -- DELEGATION_BEGIN
+  :end-before: -- DELEGATION_END
+
+.. literalinclude:: daml/4_Transformations/Contact.daml
+  :language: daml
+  :start-after: -- DELEGATION_TEST_BEGIN
+  :end-before: -- DELEGATION_TEST_END
+
+If you open the scenario view in the IDE, you will notice that Bob sees the ``Contact``. Controllers specified via ``controller c can`` syntax become *observers* of the contract. More on *observers* later, but in short, they get to see any changes to the contract.
+
+Choices in the Ledger Model
+---------------------------
+
+It's time for a bit more theory. in :doc:`1_Token` you learned about the high-level structure of a DAML ledger. With choices and the `exercise` function, you have the next important ingredient. A *transaction* is a list of *actions*, and there are just three kinds of action: ``create``, ``exercise`` and ``fetch``. All actions are performed on a contract.
```

Comment:
> It's time for a bit more theory.

Why? What will I get out of it? ie - could you add something about what this will help you achieve/understand that you couldn't before? Otherwise this is quite offputting.

Reply:
Who doesn't love theory!?

Reply:
```suggestion
In :doc:`1_Token` you learned about the high-level structure of a DAML ledger. With choices and the `exercise` function, you have the next important ingredient to understand the structure of the ledger and transactions. A *transaction* is a list of *actions*, and there are just three kinds of action: ``create``, ``exercise`` and ``fetch``. All actions are performed on a contract.
```